x-pack/metricbeat/module/azure/billing/service.go

```diff
@@ -85,46 +116,83 @@ func (service *UsageService) GetForecast(scope string, startTime, endTime time.T
-		Dataset: &costmanagement.QueryDataset{
-			Aggregation: map[string]*costmanagement.QueryAggregation{
+	granularityDaily := armcostmanagement.GranularityTypeDaily
+
+	forecastTimeframeCustom := armcostmanagement.ForecastTimeframeTypeCustom
+	forecastTypeActualCost := armcostmanagement.ForecastTypeActualCost
+
+	forecastDefinition := armcostmanagement.ForecastDefinition{
+		Dataset: &armcostmanagement.ForecastDataset{
+			Aggregation: map[string]*armcostmanagement.QueryAggregation{
 				"totalCost": {
-					Function: aggregationFunction,
+					Function: &aggregationFunction,
 					Name:     &aggregationName,
 				},
 			},
-			Granularity: costmanagement.GranularityTypeDaily,
+			Granularity: &granularityDaily,
 		},
 
 		// Time frame/period of the forecast. Required for MCA accounts.
 		//
 		// If omitted, EA users will get a forecast for the current month, and
 		// MCA users will get an error.
-		Timeframe: costmanagement.ForecastTimeframeTypeCustom,
-		TimePeriod: &costmanagement.QueryTimePeriod{
-			From: &date.Time{Time: startTime},
-			To:   &date.Time{Time: endTime},
+		Timeframe: &forecastTimeframeCustom,
+		TimePeriod: &armcostmanagement.QueryTimePeriod{
+			From: &startTime,
+			To:   &endTime,
 		},
 
-		Type:                    costmanagement.ForecastTypeActualCost,
+		Type:                    &forecastTypeActualCost,
 		IncludeActualCost:       &includeActualCost,
 		IncludeFreshPartialCost: &includeFreshPartialCost,
 	}
 
 	// required, but I don't have a use for it, yet.
 	filter := ""
 
-	queryResult, err := service.forecastClient.Usage(service.context, scope, forecastDefinition, filter)
+	queryResult, err := service.forecastClient.Usage(service.context, scope, forecastDefinition, &armcostmanagement.ForecastClientUsageOptions{
+		Filter: &filter,
+	})
 	if err != nil {
-		return costmanagement.QueryResult{}, err
+		return armcostmanagement.QueryResult{}, err
 	}
 
-	return queryResult, nil
+	return queryResult.QueryResult, nil
 }
 
 // GetUsageDetails fetches the usage details for the given filters.
-func (service *UsageService) GetUsageDetails(scope string, expand string, filter string, skipToken string, top *int32, metrictype consumption.Metrictype, startDate string, endDate string) (consumption.UsageDetailsListResultPage, error) {
-	return service.usageDetailsClient.List(service.context, scope, expand, filter, skipToken, top, metrictype, startDate, endDate)
+func (service *UsageService) GetUsageDetails(
+	scope string,
+	expand string,
+	filter string,
+	skipToken string,
+	top *int32,
+	metrictype armconsumption.Metrictype,
+	startDate string,
+	endDate string,
+) (armconsumption.UsageDetailsListResult, error) {
+	pager := service.usageDetailsClient.NewListPager(scope, &armconsumption.UsageDetailsClientListOptions{
+		Expand: &expand,
+		Filter: &filter,
+		Metric: &metrictype,
+		// Skiptoken: &skipToken,
+		// Top:       top,
+		StartDate: &startDate,
+		EndDate:   &endDate,
+	})
```

Comment:
As mentioned above, we can probably eliminate the `skipToken` and `top` arguments if we don't use them.